src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterFactory.cs

```diff
@@ -14,6 +14,8 @@ namespace System.Text.Json.Serialization
     /// </remarks>
     public abstract class JsonConverterFactory : JsonConverter
     {
+        internal const string JsonConverterFactoryRequiresDynamicCodeMessage = "Types deriving from JsonConverterFactory cannot be statically analyzed and might require runtime code generation.";
```

Comment:
Is this accurate statement? It should be possible to implement statically analyzable AOT friendly types that inherit from JsonConverterFactory.

Reply:
> It should be possible to implement statically analyzable AOT friendly types that inherit from JsonConverterFactory.

Possibly, but I would think that only applies to trivial cases that don't constitute the typical use case for converter factories; `Type.MakeGenericType` is the bread and butter for this abstraction. 

That being said, only `JsonStringEnumConverter` is being annotated with this warning, not the base `JsonConverterFactory` class. The const string is being hosted here for technical reasons, see https://github.com/dotnet/runtime/pull/73161#discussion_r934726896.